src/lib.rs

```diff
@@ -11,6 +11,898 @@
+                let instance_id = 1;
+                let (_, event_id) = split_u32_to_u16(source_filter.resource_id);
+
+                trace!(
+                    "{}:{} - register_message_handler: service: {} instance: {} method: {}",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                    service_id,
+                    instance_id,
+                    event_id
+                );
+
+                {
+                    let requested_events = REQUESTED_EVENTS.write().await;
+                    if !requested_events.contains(&(service_id, instance_id, event_id)) {
+                        get_pinned_application(application_wrapper).request_service(
+                            service_id,
+                            instance_id,
+                            ANY_MAJOR,
+                            vsomeip::ANY_MINOR,
+                        );
+                        request_single_event_safe(
+                            application_wrapper,
+                            service_id,
+                            instance_id,
+                            event_id,
+                            event_id,
+                        );
+                        get_pinned_application(application_wrapper).subscribe(
+                            service_id,
+                            instance_id,
+                            event_id,
+                            ANY_MAJOR,
+                            event_id,
+                        );
+                    }
+                }
+                register_message_handler_fn_ptr_safe(
+                    application_wrapper,
+                    service_id,
+                    instance_id,
+                    event_id,
+                    msg_handler,
+                );
+
+                // Letting vsomeip settle
+                tokio::time::sleep(Duration::from_millis(5)).await;
+
+                trace!(
+                    "{}:{} - Registered vsomeip message handler.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+
+                Ok(())
+            }
+            RegistrationType::Request(_) => {
+                trace!(
+                    "{}:{} - Registering for Request style messages.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+                let Some(sink_filter) = sink_filter else {
+                    return Err(UStatus::fail_with_code(
+                        UCode::INVALID_ARGUMENT,
+                        "Unable to map source and sink filters to uProtocol message type",
+                    ));
+                };
+
+                let (_, service_id) = split_u32_to_u16(sink_filter.ue_id);
+                let instance_id = 1; // TODO: Set this to 1? To ANY_INSTANCE?
```

Comment:
What should this be set to? Wondering if you know @int0x27 / @stevenhartley

Reply:
Instance ID will eventually be defined in some uService artifact file and then fed intot he static JSON that SOME/IP needs to function but for now we assume 1.

Reply:
use high 16 bits of `ue_id` as instance. If they are `0x0000`, assume `instance_id=1`